rerun_cpp/CMakeLists.txt

```diff
@@ -11,64 +11,115 @@ file(GLOB_RECURSE rerun_sdk_SRC CONFIGURE_DEPENDS
 
 add_library(rerun_sdk ${rerun_sdk_SRC})
 
-set_default_warning_settings(rerun_sdk)
+# Rerun needs C++17.
+set_property(TARGET rerun_sdk PROPERTY CXX_STANDARD 17)
 
 # ------------------------------------------------------------------------------
+# Rerun development setup only:
+if(DEFINED RERUN_REPOSITORY)
+    # execute process doesn't fail if the process fails.
+    # `COMMAND_ERROR_IS_FATAL ANY` parameter fixes this but is only available in CMake 3.19
+    # which isn't default on Ubuntu LTS as of writing, causing unnecessary friction.
+    # TODO(andreas): use add_custom_command instead so this runs at build time! https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command
+    execute_process(COMMAND cargo build --release -p re_types RESULT_VARIABLE ret) # Generates most of the C++ source files
```

Comment:
```suggestion
    execute_process(COMMAND just codegen RESULT_VARIABLE ret) # Generates most of the C++ source files
```

Reply:
It makes sense, but I'm unhappy about this requiring more dependencies for people coming in to build cpp themselves

Reply:
actually I think now we should just remove this line. Codegen should *always* be explicit now